rust/lanzatool/src/generation.rs

```diff
@@ -1,47 +1,71 @@
+use serde::{Deserialize, Serialize};
+
 use std::fmt;
 use std::fs;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 use anyhow::{Context, Result};
+use bootspec::BootJson;
+use bootspec::SpecialisationName;
```

Comment:
Do you know why they used wrapper types upstream? Looks kinda arbitrary that these are newtypes while other parts of the spec aren't

Reply:
invoking @cole-h regarding this

Reply:
:shrug: 

It lets people design APIs that only wants to operate on a specialisation via its name and leverages the type system for that.

Or something like that; I don't remember the exact reasoning, since I wrote that part a long time ago (to me) :sweat_smile:

Reply:
That's what I guessed was the idea. In my opinion we should either make all fields newtypes with a `::new` constructor, so that we actually hide the implementation details or turn all fields into the raw "underlying" type (i.e. `SpecialisationName` becomes  `String`). We don't have to solve this here, but It would be nice to eventually address it upstream.

Reply:
@nikstur Please open an issue upstream and we will track this there